sdk/src/main/scala/com/horizen/certificatesubmitter/CertificateSubmitter.scala

```diff
@@ -564,7 +571,20 @@ class CertificateSubmitter(settings: SidechainSettings,
       signersPublicKeyWithSignatures)
   }
 
+  private def lastSidechainEpochNumberForWithdrawalEpochNumber(history: SidechainHistory, withdrawalEpochNumber: Int): ConsensusEpochNumber = {
+    val headerInfo: MainchainHeaderInfo = getLastMainchainBlockHashForWithdrawalEpochNumber(history, withdrawalEpochNumber)
+
+    val parentBlockInfo: SidechainBlockInfo = history.storage.blockInfoById(headerInfo.sidechainBlockId)
+    TimeToEpochUtils.timeStampToEpochNumber(params, parentBlockInfo.timestamp)
+  }
+
   private def lastMainchainBlockCumulativeCommTreeHashForWithdrawalEpochNumber(history: SidechainHistory, withdrawalEpochNumber: Int): Array[Byte] = {
+    val headerInfo: MainchainHeaderInfo = getLastMainchainBlockHashForWithdrawalEpochNumber(history, withdrawalEpochNumber)
+
+    headerInfo.cumulativeCommTreeHash
+  }
+
+  private def getLastMainchainBlockHashForWithdrawalEpochNumber(history: SidechainHistory, withdrawalEpochNumber: Int) = {
```

Comment:
Please change name getLastMainchainBlockHashForWithdrawalEpochNumber to getLastMainchainBlock**Info**ForWithdrawalEpochNumber. The current name is a little bit confusing.